SofaKernel/modules/SofaDeformable/PolynomialSpringsForceField.inl

```diff
@@ -0,0 +1,598 @@
+/******************************************************************************
+*       SOFA, Simulation Open-Framework Architecture, version 1.0 RC 1        *
+*                (c) 2006-2020 MGH, INRIA, USTL, UJF, CNRS                    *
+*                                                                             *
+* This library is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This library is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this library; if not, write to the Free Software Foundation,     *
+* Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301 USA.          *
+*******************************************************************************
+*                               SOFA :: Modules                               *
+*                                                                             *
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+
+#include <sofa/core/behavior/ForceField.inl>
+#include "PolynomialSpringsForceField.h"
+#include <sofa/core/visual/VisualParams.h>
+#include <sofa/helper/system/config.h>
+#include <sofa/defaulttype/VecTypes.h>
+#include <sofa/defaulttype/RigidTypes.h>
+#include <sofa/helper/gl/template.h>
+#include <sofa/simulation/AnimateBeginEvent.h>
+#include <assert.h>
+#include <math.h>
+#include <iostream>
+#include <fstream>
+#include <sofa/helper/AdvancedTimer.h>
+
+
+namespace sofa
+{
+
+namespace component
+{
+
+namespace interactionforcefield
+{
+
+
+template<class DataTypes>
+PolynomialSpringsForceField<DataTypes>::PolynomialSpringsForceField()
+    : PolynomialSpringsForceField(nullptr, nullptr)
+{
+}
+
+
+template<class DataTypes>
+PolynomialSpringsForceField<DataTypes>::PolynomialSpringsForceField(MechanicalState* mstate1, MechanicalState* mstate2)
+    : Inherit(mstate1, mstate2)
+    , d_firstObjectPoints(initData(&d_firstObjectPoints, "firstObjectPoints", "points related to the first object"))
```

Comment:
Why not use the same naming as the ones used in SpringsForceField ? Renaming things is the best way to annoy users.